src/utils/fromPairs.ts

```diff
@@ -0,0 +1,15 @@
+export default function fromPairs<TValue>(iterable: Iterable<[string, TValue]>)
```

Comment:
Whats this complicated logic about? The loop issue is because of this function. If my understanding is correct, the idea is to get an object of of key, val pairs. Its a simple array.reduce function away. Why have complicated difficult to parse code?